tests/bot/cogs/moderation/test_utils.py

```diff
@@ -0,0 +1,477 @@
+
+                send_private_embed_mock.assert_awaited_once_with(args[0], embed)
+
+                self.ctx.send.reset_mock(side_effect=True)
+                send_private_embed_mock.reset_mock()
+
+    @patch("bot.cogs.moderation.utils.send_private_embed")
+    async def test_notify_pardon(self, send_private_embed_mock):
+        """Test does `notify_pardon` create correct result."""
+        test_cases = [
+            {
+                "args": (self.user, "Test title", "Example content"),
+                "expected_output": {
+                    "description": "Example content",
+                    "title": "Test title",
+                    "icon_url": Icons.user_verified
+                },
+                "send_result": True,
+                "send_raise": None
+            },
+            {
+                "args": (self.user, "Test title 1", "Example content 1", Icons.user_update),
+                "expected_output": {
+                    "description": "Example content 1",
+                    "title": "Test title 1",
+                    "icon_url": Icons.user_update
+                },
+                "send_result": False,
+                "send_raise": NotFound(AsyncMock(), AsyncMock())
+            }
+        ]
+
+        for case in test_cases:
+            args = case["args"]
+            expected = case["expected_output"]
+            send, send_raise = case["send_result"], case["send_raise"]
+
+            with self.subTest(args=args, expected=expected):
+                if send_raise:
+                    self.ctx.send.side_effect = send_raise
+
+                send_private_embed_mock.return_value = send
+
+                result = await notify_pardon(*args)
+
+                self.assertEqual(send, result)
+
+                embed = send_private_embed_mock.call_args[0][1]
+
+                self.assertEqual(embed.description, expected["description"])
+                self.assertEqual(embed.colour.value, PARDON_COLOR)
+                self.assertEqual(embed.author.name, expected["title"])
+                self.assertEqual(embed.author.icon_url, expected["icon_url"])
+
+                send_private_embed_mock.assert_awaited_once_with(args[0], embed)
+
+                self.ctx.send.reset_mock(side_effect=True)
+                send_private_embed_mock.reset_mock()
+
+    async def test_post_user(self):
+        """Test does `post_user` work correctly."""
+        test_cases = [
+            {
+                "args": (self.ctx, self.user),
+                "post_result": [
+                    {
+                        "id": 1234,
+                        "avatar": "test",
+                        "name": "Test",
+                        "discriminator": 1234,
+                        "roles": [
+                            1234,
+                            5678
+                        ],
+                        "in_guild": False
+                    }
+                ],
```

Comment:
First, this is constant across test cases. Second, there's no need for it to be emulating a real-world value. `post_user` just returns whatever `api_client.post` returns. Because you mock the latter, you control its return value. You just need to assert that the return value of `post_user` equals the return value of `api_client.post` e.g.:

```py
expected = "dummy value"

self.ctx.bot.api_client.post.return_value = expected
result = await post_user(*args)
self.assertEqual(result, expected)
```

The above achieves the same effect.